File: python/selectionSort.py
#encoding: utf-8

def findSmallest(arr):
    smallest = arr[0]
    small_index = 0
    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            small_index = i
    return small_index

def selectionSort(arr):
    newArr = []
    for i in range(len(arr)):
        smallest = findSmallest(arr)
        print smallest
        newArr.append(arr.pop(smallest))
    return newArr
        

# test
print selectionSort([2,4,6,1,9,7,5])

File: php/selectionSort.php
<?php
function selectionSort($arr){

    $len = count($arr);

    for ($i = 0; $i < $len - 1; $i++) {
        $smallest = $arr[i];
        $small_index = i;
        for ($j = i + 1; $j < $len; $j++) {
            if ($smallest > $arr[j]) {
                $smallest = $arr[j];
                $small_index = j;
            }
        }
        $temp = $arr[i];
        $arr[j] = $temp;
        $arr[i] = $smallest;
    }
}

?>
